### ISSUE TITLE
[Bug]: How to remove extra spaces from i-text for non-English languages like hindi, gujarati, etc?

### ISSUE THREAD

**Issue by BhargavZanzmera** [fabricjs/fabric.js#9357]
### CheckList

- [x] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

5.3.0

### In What environments are you experiencing the problem?

Firefox, Chrome, Safari, Opera, Microsoft Edge

### Node Version (if applicable)

None

### Link To Reproduction

http://fabricjs.com/test/misc/itext.html

### Steps To Reproduce

1. Copy the word उज्जैन or any other hindi word that includes joint character.
2. Paste it in itext object of above given link.
3. You can see after word ends there is an extra space. Because of that it creates curser issue also.

Note : watch below video to reproduce bug. If you are unable to create it then try some other joint word of hindi or gujarati.

[screen-capture (4).webm](https://github.com/fabricjs/fabric.js/assets/145420691/434398be-c9aa-4039-8f30-e13307183619)

### Expected Behavior

I-text box size should be same as per the content it has.

### Actual Behavior

![hindi 2](https://github.com/fabricjs/fabric.js/assets/145420691/29476773-02f1-415c-9d5e-cc62db06b3b5)
![Annotation 2](https://github.com/fabricjs/fabric.js/assets/145420691/b418bfd6-a2d7-4931-bd22-33aefa26b80b)

**Comment by asturur** [comment]
Fabric grapheme abilities are not enough for hindi.
<img width="319" alt="image" src="https://github.com/fabricjs/fabric.js/assets/1194048/962ada5d-0a71-40ca-9ba5-ea9a14e43041">

i suggest you take a grapheme splitting library like this one that covers unicode up to 15 https://github.com/orisano/graphemesplit
and override fabric.util.string.graphemeSplit.

As state in docs fabric.util.string.graphemeSplit is meant to be changed with specialized libraries, since we won't be updating it to run behind unicode changes.

**Comment by vimlesh1975** [comment]
installed npm i graphemesplit, then
I used like this and problem solved.
`import split from 'graphemesplit'
fabric.util.string.graphemeSplit = split`

**Comment by ShaMan123** [comment]
In v6 you need to override FabricText

**Comment by ShaMan123** [comment]
And maybe it is time it is part of the repo

**Comment by asturur** [comment]
Yes in 6.6.5 for browser that support the international segmenter this issue is solved.
